x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/common/ProgressListenableActionFuture.java

```diff
@@ -25,11 +24,13 @@
  */
 class ProgressListenableActionFuture extends PlainActionFuture<Long> {
 
+    private record PositionAndListener(Long position, ActionListener<Long> listener) {}
```

Comment:
Nit: I believe `position` can be a primitive `long`?

Reply:
Yes, I pushed 9b857ab83d5f0cba3034d0b85a10e796fe8567ed. Thanks!